glmmbernoulli.qmd

```diff
@@ -0,0 +1,523 @@
+as in @fig-contradata,
+
+```{julia}
+#| code-fold: true
+#| fig-cap: Smoothed relative frequency of contraception use versus centered age for women in the 1989 Bangladesh Fertility Survey
+#| label: fig-contradata
+draw(
+  data(contra) *
+  mapping(
+    :age => "Centered age (yr)",
+    :use => ==("Y") => "Frequency of contraceptive use";
+    col=:urban => renamer(["N" => "Rural", "Y" => "Urban"]),
+    color=:livch,
+  ) *
+  smooth();
+  figure=(; resolution=(800, 450)),
+)
+```
+
+to show the trend in the response with respect to the covariate.
+Once we have the smoother lines in such a plot we can omit the data points themselves, as we did here, because they add very little information.
+
+The first thing to notice about the plot is that the proportion of women using contraception is not linear in age, which, on reflection, makes sense.
+A woman in the middle of this age range (probably corresponding to an age around 25) is more likely to use artificial contraception than is a girl in her early teens or a woman in her mid-forties.
+We also see that women in an urban setting are more likely to use contraception than those in a rural setting and that women with no live children are less likely to use contraception than are women who already have children.
+There do not seem to be strong differences between women who have 1, 2 or 3 or more children compared to the differences between women with children and those without children.
+
+Interestingly, the quadratic pattern with respect to age does not seem to have been noticed in other analyses of these data.
+Comparisons of model fits through different software systems, as provided by the Center for Multilevel Modelling, incorporate only a linear term in age, even though the pattern is clearly nonlinear.
+The lesson here is similar to what we have seen in other examples; careful plotting of the data should, whenever possible, precede attempts to fit models to the data.
+
+### Initial GLMM fit to the contraception data {#sec-contraglmm}
+
+Fitting a generalized linear mixed model (GLMM) is very similar to fitting a linear mixed model (LMM).
+We call the `fit` function with the first three arguments being the model type, `MixedModel`, then a formula specifying the response, fixed-effects terms and random-effects terms, then a data table.
+For GLMMs we also specify a fourth positional argument which is a distribution - in this case `Bernoulli()`.
+
+Establishing the contrasts and fitting a preliminary model with random effects for `dist`, main effects for `livch`, `age`, `age^2`, and `urban`, plus interaction terms for `age & urban` and `age^2 & urban` can be done as
+
+```{julia}
+contrasts[:livch] = EffectsCoding(; base="0")
+contrasts[:urban] = HelmertCoding()
+contrasts[:dist] = Grouping()
+
+com01 = let
+  form = @formula(use ~ 1 + livch + age * age * urban + (1 | dist))
```

Comment:
I would probably take advantage of the distributive property rather than slightly abusing an interaction term:

```suggestion
  form = @formula(use ~ 1 + livch + (age + abs2(age)) * urban + (1 | dist))
```